From b0e084027f528bbc42bd4c5a19c6ec441518044a Mon Sep 17 00:00:00 2001
From: Susant Sahani <ssahani@redhat.com>
Date: Tue, 10 Feb 2026 15:15:27 +0530
Subject: [PATCH] output: sanitize guest names in metadata file paths

Use sanitize_slash for metadata filenames to avoid '/' in paths.

Fixes: https://issues.redhat.com/browse/RHEL-136479
Signed-off-by: Susant Sahani <ssahani@redhat.com>
(cherry picked from commit e02563cfcb0022cdd3a044c6a8e459eeb2c5abf6)
---
 output/output_disk.ml     | 2 +-
 output/output_kubevirt.ml | 2 +-
 output/output_qemu.ml     | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/output/output_disk.ml b/output/output_disk.ml
index af0a3fac..dc43d637 100644
--- a/output/output_disk.ml
+++ b/output/output_disk.ml
@@ -105,7 +105,7 @@ module Disk = struct
                 (disk_path output_storage output_name)
                 output_format output_name in
 
-    let file = output_storage // output_name ^ ".xml" in
+    let file = output_storage // (sanitize_slash output_name) ^ ".xml" in
     with_open_out file (fun chan -> DOM.doc_to_chan chan doc);
 
     if verbose () then (
diff --git a/output/output_kubevirt.ml b/output/output_kubevirt.ml
index 8a5ab56d..8b011436 100644
--- a/output/output_kubevirt.ml
+++ b/output/output_kubevirt.ml
@@ -146,7 +146,7 @@ module Kubevirt = struct
     let doc = create_kubevirt_yaml source inspect target_meta disk_path
                 output_format output_name in
 
-    let file = output_storage // output_name ^ ".yaml" in
+    let file = output_storage // (sanitize_slash output_name) ^ ".yaml" in
     with_open_out file (fun chan -> YAML.doc_to_chan chan doc);
 
     if verbose () then (
diff --git a/output/output_qemu.ml b/output/output_qemu.ml
index 63299423..b9090d86 100644
--- a/output/output_qemu.ml
+++ b/output/output_qemu.ml
@@ -111,7 +111,7 @@ module QEMU = struct
     (* Start the shell script.  Write it to a temporary file
      * which we rename at the end.
      *)
-    let file = output_storage // output_name ^ ".sh" in
+    let file = output_storage // (sanitize_slash output_name) ^ ".sh" in
     let tmpfile = file ^ ".tmp" in
     On_exit.unlink tmpfile;
 
